chain/chain/src/chain.rs

```diff
@@ -3129,17 +3242,31 @@ impl<'a> ChainUpdate<'a> {
             }
         }
 
+        let mut apply_chunk_work = Vec::new();
+
         // Always apply state transition for shards in the current epoch
-        self.apply_chunks(me, block, &prev_block, ApplyChunksMode::ThisEpoch)?;
+        apply_chunk_work.extend(self.apply_chunks_preprocessing(
+            me,
+            block,
+            &prev_block,
+            ApplyChunksMode::ThisEpoch,
+        )?);
```

Comment:
Just as an alternative, one pattern I like when I need to assemble a complex sequence of things is passing an accumulator into the function, instead of retruning a vec:

```rust
fn before() -> Vec<Thing>;
fn after(acc: &mut Vec<Thing>);
```

This allows saving intermediate allocations, but is mostly unrelated to this PR.